### PR TITLE
[COM-231] Adds jest to middleware project

### PR DESCRIPTION
**Pull request by hannahtrask**
## Summary

 - Adds Jest to the NextJS middleware project
 - Requires naming of test files to follow a pattern `**util.u.spec.**`

### CODE REVIEW

File: src/function.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export default function helloWorld() {
```
Comment (hannahtrask) [inline src/function.ts]: Just a function to test that running `npm run test` works with Typescript files. Can be deleted.

File: jest.config.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+const unitConfig = {
+  testRegex: ['/*.u.spec.*$'],
+  setupFilesAfterEnv: ['<rootDir>/jest.u.setup.ts'],
+  ...baseConfig,
+}
+
+module.exports = {
+ preset: 'ts-jest',
```
Comment (hannahtrask) [inline jest.config.js]: https://www.npmjs.com/package/ts-jest

File: package-lock.json [hunk 3]
```diff
@@ -4,6 +4,28 @@
   "lockfileVersion": 1,
   "requires": true,
   "dependencies": {
+    "@ampproject/remapping": {
```
Comment (hannahtrask) [inline package-lock.json:7]: It hurts my heart that this is such a massive change.

File: src/function.ts (hunk 1 above)
Comment (alxgmpr) [reply src/function.ts]: Fine with this but can you just inline it directly into the test and call it helloWorld.u.spec.ts?

File: jest.config.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+const baseConfig = {
+  moduleNameMapper: {
+    '@tests/(.*)': '<rootDir>/src/tests/$1',
+  },
+  roots: ['src'],
+}
+
+const unitConfig = {
+  testRegex: ['/*.u.spec.*$'],
+  // setupFilesAfterEnv: ['<rootDir>/jest.u.setup.ts'],
```
Comment (alxgmpr) [inline jest.config.js]: This comment can be removed

File: jest.u.setup.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+import '@testing-library/jest-dom'
```
Comment (alxgmpr) [inline jest.u.setup.ts]: I don't think this is necessary (and by proxy don't think this setup file is necessary -- at least yet). We aren't doing any DOM testing. Note that our `testEnvironment` is `node` not `browser`

Also this package isn't in your `package.json` anyway...

File: package-lock.json (hunk 3 above)
Comment (alxgmpr) [reply package-lock.json:7]: Fine for there to be a big change when adding a package like Jest but you've included TryNow-internal CodeArtifact URLs in this diff here that won't be usable by the general public.

You need to revert and re-generate `package-lock.json` but without the CodeArtifact configuration set (`npm config delete registry` I think)

**Review by alxgmpr — CHANGES_REQUESTED** [review]

**Review by alxgmpr — APPROVED** [review]
Looks good to me 👍

**Review by joshrathke — APPROVED** [review]

**Pull request merged**